app/controllers/content/live_pages_reports_controller.rb

```diff
@@ -0,0 +1,33 @@
+class Content::LivePagesReportsController < ApplicationController
+  def show
+    relation = Facts::Metric
+                 .joins(:dimensions_date)
+                 .group(:date_name)
+
+    if organisation_filter.present?
```

Comment:
Could you please work towards extracting this logic out of the controller? Something simple like:

```ruby
@report = Reports::LivePages.run(organisation: params[organisation))
```

Reply:
I've moved the query logic to scopes on `Facts::Metric` so now the controller just does:

```rb
Facts::Metric
  .by_date_name
  .for_organisation(content_id: filter_params[:organisation])
  .count
```

We could go further and extract the combination of the two scopes and the call to `count` into another object but I feel like that's just adding more layers at this time.